python/lsst/afw/image/image/fitsIoWithOptions.py

```diff
@@ -60,22 +64,24 @@ def imageReadFitsWithOptions(cls, source, options):
         If options contains "llcX" and is missing any of
         "llcY", "width", or "height".
     """
+    origin = ImageOrigin.PARENT
     bbox = lsst.geom.Box2I()
-    if options.exists("llcX"):
+    if "bbox" in options.keys():
+        bbox = options["bbox"]
+    elif isinstance(options, PropertySet) and options.exists("llcX"):
```

Comment:
This should support `dict` and not be restricted to PropertySet. You can use `in` so this will work just like the check for `bbox` above:

```python
elif "llcx" in options:
```